src/dlgConnectionProfiles.cpp

```diff
@@ -1829,41 +1804,39 @@ bool dlgConnectionProfiles::validateProfile()
         check.setHost(url);
         if (!check.isValid()) {
             notificationAreaIconLabelError->show();
-            notificationAreaMessageBox->setText(QString("%1\n%2").arg(notificationAreaMessageBox->text(), tr("Please enter the URL or IP address of the Game server.\n\n%1").arg(check.errorString())));
+            notificationAreaMessageBox->setText(QString("%1\n%2").arg(notificationAreaMessageBox->text(), tr("Please enter the URL of the Game server.%1").arg(check.errorString())));
             host_name_entry->setPalette(mErrorPalette);
-            validUrl = false;
-            valid = false;
+            validProfile = false;
         }
 
-        if (url.indexOf(QRegularExpression(QStringLiteral("^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])$")), 0) != -1) {
-            if (port_ssl_tsl->isChecked()) {
+        if (port_ssl_tsl->isChecked()) {
+            if (url.indexOf(QRegularExpression(QStringLiteral("^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$")), 0) != -1) {
```

Comment:
:bulb: Although **not** something for this PR, if this is what I think it is, namely a check for a *dot-separated numeric IPv4 address* then it is not enough here - as it does not prevent a ***colon-separated IPv6 address***!